docs/static/transforming-data.asciidoc

```diff
@@ -0,0 +1,520 @@
++
+[source,json]
+--------------------------------------------------------------------------------
+filter {
+  dns {
+    reverse => [ "source_host" ]
+    action => "replace"
+  }
+}
+--------------------------------------------------------------------------------
+
+    
+<<plugins-filters-elasticsearch,elasticsearch>>::
+
+Copies fields from previous log events in Elasticsearch to current events. 
++
+[source,json]
+--------------------------------------------------
+NEED A SIMPLE EXAMPLE HERE. 
+--------------------------------------------------
+
+
+//REVEIWERS: I NEED A BETTER EXAMPLE FOR THE ELASTICSEARCH FILTER. THE EXAMPLE IN THE DOCUMENTATION MIGHT BE TOO COMPLEX FOR AN OVERVIEW.
+
+
+<<plugins-filters-geoip,geoip filter>>::
+
+Adds geographical information about the location of IP addresses. For example: 
++
+[source,json]
+--------------------------------------------------------------------------------
+filter {
+  geoip {
+    source => "clientip"
+  }
+}
+--------------------------------------------------------------------------------
++    
+After the geoip filter is applied, the event will be enriched with geoip fields.
+For example:
++
+[source,json]
+--------------------------------------------------------------------------------
+          "geoip" => {
+              "timezone" => "Europe/Moscow",
+                    "ip" => "83.149.9.216",
+              "latitude" => 55.7522,
+        "continent_code" => "EU",
+             "city_name" => "Moscow",
+         "country_code2" => "RU",
+          "country_name" => "Russia",
+              "dma_code" => nil,
+         "country_code3" => "RU",
+           "region_name" => "Moscow",
+              "location" => [
+            [0] 37.6156,
+            [1] 55.7522
+        ],
+           "postal_code" => "101194",
+             "longitude" => 37.6156,
+           "region_code" => "MOW"
+    }
+--------------------------------------------------------------------------------
+
+<<plugins-filters-jdbc_streaming,jdbc_streaming>>::
+
+Enriches events with database data.
++
+The following example executes a SQL query and stores the result set in a field
+called `country_details`:
++
+[source,json]
+--------------------------------------------------------------------------------
+filter {
+  jdbc_streaming {
+    jdbc_driver_library => "/path/to/mysql-connector-java-5.1.34-bin.jar"
+    jdbc_driver_class => "com.mysql.jdbc.Driver"
+    jdbc_connection_string => ""jdbc:mysql://localhost:3306/mydatabase"
+    jdbc_user => "me"
+    jdbc_password => "secret"
+    statement => "select * from WORLD.COUNTRY WHERE Code = :code"
+    parameters => { "code" => "country_code"}
+    target => "country_details"
+  }
+}
+--------------------------------------------------------------------------------
+
+
+//REVIEWERS: THE MYSQL CONNECTOR SHOWN IN THIS EXAMPLE IS KIND OF OLD. NOT SURE I SHOULD CHANGE IT, THOUGH, BECAUSE I'M NOT SURE IF WE SUPPORT ALL VERSIONS OF THE CONNECTOR.
```

Comment:
I don't know if we specify what connectors (or versions of a connector) are supported. @guyboertje maybe has ideas?